packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -762,6 +767,78 @@ class _DialPainter extends CustomPainter {
     canvas.restore();
   }
 
+  static const double _kSemanticNodeSizeScale = 1.5;
+
+  /// Creates semantics nodes for the hour/minute labels painted on the dial.
+  ///
+  /// The nodes are positioned on top of the text and their size is
+  /// [_kSemanticNodeSizeScale] bigger than those of the text boxes to provide
+  /// bigger tap area.
+  List<SemanticsNode> graftSemantics(RenderBox renderBox) {
+    final Size size = renderBox.size;
+    final double radius = size.shortestSide / 2.0;
+    final Offset center = new Offset(size.width / 2.0, size.height / 2.0);
+    const double labelPadding = 24.0;
+    final double outerLabelRadius = radius - labelPadding;
+    final double innerLabelRadius = radius - labelPadding * 2.5;
+
+    Offset getOffsetForTheta(double theta, _DialRing ring) {
+      double labelRadius;
+      switch (ring) {
+        case _DialRing.outer:
+          labelRadius = outerLabelRadius;
+          break;
+        case _DialRing.inner:
+          labelRadius = innerLabelRadius;
+          break;
+      }
+      return center + new Offset(labelRadius * math.cos(theta),
+          -labelRadius * math.sin(theta));
+    }
+
+    final List<SemanticsNode> nodes = <SemanticsNode>[];
+
+    void paintLabels(List<TextPainter> labels, _DialRing ring) {
+      if (labels == null)
+        return;
+      final double labelThetaIncrement = -_kTwoPi / labels.length;
+      double labelTheta = math.PI / 2.0;
+
+      for (TextPainter label in labels) {
+        final double width = label.width * _kSemanticNodeSizeScale;
+        final double height = label.height * _kSemanticNodeSizeScale;
+        final Offset nodeOffset = getOffsetForTheta(labelTheta, ring) + new Offset(-width / 2.0, -height / 2.0);
+        final SemanticsNode node = new SemanticsNode(showOnScreen: renderBox.showOnScreen)
+          ..rect = new Rect.fromLTRB(
+            nodeOffset.dx,
+            nodeOffset.dy,
+            nodeOffset.dx + width,
+            nodeOffset.dy + height
+          )
+          ..updateWith(
+              config: new SemanticsConfiguration()
+                ..label = label.text.text
+                ..textDirection = textDirection
+                ..addAction(SemanticsAction.tap, () {
+                  // The action is sent to the gesture detector surrounding the
+                  // custom painter.
```

Comment:
is that supposed to happen?

cc @goderbauer

Reply:
I'll wait for @goderbauer's OK before submitting, but I checked a couple of things:

- instrumented `Window.onPointerDataPacket` and made sure the OS sends taps to the selected rectangle irrespective of where you double-tap, which tells me that the OS gets the semantic nodes created by the time picker.
- checked with other widgets (e.g. `CupertinoSwitch`) and found that they behave the same way, i.e. the closure passed to `addAction` is not invoked, and instead is sent to the gesture recognizer.

Reply:
This sounds like a bug. I am investigating...

Reply:
Turns out: The observed behavior is not a bug, but a feature in Android prior to Oreo, see https://github.com/flutter/flutter/issues/13219 for discussion.

Having said that, to make the widget also accessible for newer Android versions (Oreo and newer) as well as iOS, this handler needs to implement the actual logic for `SemanticsAction.tap`.